src/stream/src/executor/source/source_backfill_executor.rs

```diff
@@ -518,27 +529,45 @@ impl<S: StateStore> SourceBackfillExecutorInner<S> {
                                         Mutation::Pause => {
                                             // pause_reader should not be invoked consecutively more than once.
                                             if !command_paused {
-                                                pause_reader!();
+                                                if !backfill_paused {
+                                                    pause_reader!();
+                                                }
                                                 command_paused = true;
                                             } else {
-                                                tracing::warn!(command_paused, "unexpected pause");
+                                                tracing::warn!(
+                                                    command_paused,
+                                                    backfill_paused,
+                                                    "unexpected pause"
+                                                );
                                             }
                                         }
                                         Mutation::Resume => {
                                             // pause_reader.take should not be invoked consecutively more than once.
                                             if command_paused {
-                                                backfill_stream = select_with_strategy(
-                                                    input.by_ref().map(Either::Left),
-                                                    paused_reader
-                                                        .take()
-                                                        .expect("no paused reader to resume"),
-                                                    select_strategy,
-                                                );
+                                                if !backfill_paused {
```

Comment:
Is it possible to unify these two pause condition `command_paused` and `backfill_paused`?

Reply:
Oh, it seems there is a slight difference between them.

Reply:
We can introduce some state transition, e.g., 
```rust
enum State {
	/// pause_on_startup or Mutation::Pause. Resumable by Mutation::Resume
	CommandPaused,
	/// Resumable by Mutation::StartFragmentBackfill
	BackfillPaused,
	SelfPaused,
}
```
Using multiple bools is error-prone.

Reply:
We can use a struct to encapsulate the logic, due to interdependencies between different pause states.